Past changes for review

TheaAleksovDev opened the pull request.

valentin-aleksandrov commented:
Hi, @TheaAleksovDev!
Thank you for inviting me to review your PR!

When a component doesn't receive children, we tend to use it like this:
![image](https://github.com/TheaAleksovDev/ReactConceptsPractice/assets/26555796/0bac92c1-c4ee-4c4b-bdcc-a54fed6ebe7d)

valentin-aleksandrov commented on src/components/Tasks.tsx:
```diff
@@ -2,56 +2,159 @@ import { type Task as TaskType } from "./TaskInterface";
 const Tasks = () => {
   const [tasks, setTasks] = useState<TaskType[]>([]);
+  const [openCreateTask, setOpenCreateTask] = useState(false);
+  const categories: string[] = ["work", "life", "school"];
```
TypeScript in inferring such types.
 ```suggestion
  const categories = ["work", "life", "school"];
```
Here is the proof:
![image](https://github.com/TheaAleksovDev/ReactConceptsPractice/assets/26555796/d1037087-5bc2-473a-9bbc-a6e581778968)

valentin-aleksandrov commented on src/components/CreateTask.tsx:
```diff
@@ -0,0 +1,64 @@
+
+  const handleTaskChange = (event: any) => {
+    setTask(event.target.value);
+    console.log(task);
```
```suggestion
```

valentin-aleksandrov commented on src/components/Task.tsx:
```diff
@@ -1,20 +1,46 @@
+}
+
+const Task = (props: TaskProps) => {
+  console.log(props);
```
```suggestion
```

valentin-aleksandrov commented on src/components/Tasks.tsx:
```diff
@@ -2,56 +2,159 @@ import { type Task as TaskType } from "./TaskInterface";
+  };
+
+  //not the best example of currying, couldn't think of anything better
+  const displayTasks =
+    isUrgent && selectedCategory
+      ? filterTasks(isUrgent)(selectedCategory).map((item) => {
+          return (
+            <Task
+              key={item.task}
+              {...item}
+              markCompleted={markCompleted}
+              deleteTask={deleteTask}
+            ></Task>
+          );
+        })
+      : tasks.map((item) => {
+          return (
+            <Task
+              key={item.task}
+              {...item}
+              markCompleted={markCompleted}
+              deleteTask={deleteTask}
+            ></Task>
+          );
+        });
```
```suggestion
  const filteredTasks =
    isUrgent && selectedCategory
      ? filterTasks(isUrgent)(selectedCategory)
      : tasks;

  const displayTasks = filteredTasks.map((item) => {
    return (
      <Task
        key={item.task}
        {...item}
        markCompleted={markCompleted}
        deleteTask={deleteTask}
      />
    );
  });
```

valentin-aleksandrov replied:
Notice that this way we are avoiding duplicated code.

valentin-aleksandrov commented on src/components/Task.tsx:
```diff
@@ -1,20 +1,46 @@
-          <p className="priority">{props.priority}</p>
-          <p className="category">{props.category}</p>
-        </div>}
+        <p className={`task-name ${props.isCompleted && "completed"}`}>
```
You may be interested in this library:
- https://www.npmjs.com/package/classnames

It gives you a good API to do similar things.
With that library, the above could be simplified to:
 ```suggestion
        <p className={classNames('task-name', {'completed': props.isCompleted  })}>
```

valentin-aleksandrov commented on src/components/CreateTask.tsx:
```diff
@@ -0,0 +1,64 @@
+import { useState } from "react";
+import "./tasks.css";
+
+const CreateTask = (props: any) => {
```
This component is missing types.

valentin-aleksandrov commented on src/components/Tasks.tsx:
```diff
@@ -2,56 +2,159 @@ import { type Task as TaskType } from "./TaskInterface";
-    const deleteTask = () => {
-      setTasks(tasks.filter((element) => element.task !== task));
-    };
+    let completed = false;
```
```suggestion
```

valentin-aleksandrov commented on src/components/Tasks.tsx:
```diff
@@ -2,56 +2,159 @@ import { type Task as TaskType } from "./TaskInterface";
+            <p>filter urgent tasks:</p>
+            <button
+              style={
+                isUrgent ? { backgroundColor: "#9e2b2b", color: "white" } : {}
```
Usually, react developers apply classes with predefined styles instead of using inline styles.
For the purpose of this repository, this is OK.

valentin-aleksandrov commented:
When developers apply suggestions, they must make sure the the other things are working:
![image](https://github.com/TheaAleksovDev/ReactConceptsPractice/assets/26555796/3cdd23a4-6ac1-4dcb-bd94-ab82719fab34)


classNames is a third-party library that needs to be install before using it.

TheaAleksovDev commented:
> When developers apply suggestions, they must make sure the the other things are working: ![image](https://private-user-images.githubusercontent.com/26555796/345394446-3cdd23a4-6ac1-4dcb-bd94-ab82719fab34.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwMDUyOTQsIm5iZiI6MTcyMDAwNDk5NCwicGF0aCI6Ii8yNjU1NTc5Ni8zNDUzOTQ0NDYtM2NkZDIzYTQtNmFjMS00ZGNiLWJkOTQtYWI4MjcxOWZhYjM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDExMDk1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5N2RjMzlhNjVkMWIxZGMzNDI4MWY0YWViYTJlMmQzNzM0ZjNiZjFkMzVjMmYyZjcwNWU0YTUxNGEyODUxNTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.o4Fw00MJCk3qKzWkC_h7X5Nr0_q9UiA6T2w8W2B9M2U)
> 
> classNames is a third-party library that needs to be install before using it.

fixed it and added types
